crates/bevy_ecs/src/query/iter.rs

```diff
@@ -534,13 +539,21 @@ impl<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> QueryIterationCursor<'w, 's,
             last_change_tick,
             change_tick,
         );
+        let table_entities: &[Entity] = &[];
+        let archetype_entities: &[ArchetypeEntity] = &[];
         QueryIterationCursor {
             fetch,
             filter,
-            table_entities: &[],
-            archetype_entities: &[],
-            table_id_iter: query_state.matched_table_ids.iter(),
-            archetype_id_iter: query_state.matched_archetype_ids.iter(),
+            id_iter: if Self::IS_DENSE {
```

Comment:
This should be a match statement to make adding new storage types less error prone.

Reply:
IMO we should actually probably replace the `IS_DENSE` bool with an enum?

Reply:
It'd make the const evals for IS_DENSE for fetches notably harder to read without the current boolean composition.

Also I thought we were avoiding `match bool`?

Reply:
Yeah: that's part of why I suggested an enum ;)

I agree on the const evals though, so I'm fine to leave it. Adding new storage types will be very rare, and even then they'll likely be meaningfully characterized as dense or not.